en/02_Developer_Guides/14_Files/05_File_Manipulation.md

```diff
@@ -0,0 +1,419 @@
+above methods (after content and filename). The available constants are:
+
+| Constant                            | If an existing file is found then:  |
+| ----------------------------------- | ----------------------------------- |
+| `AssetStore::CONFLICT_EXCEPTION`    | An exception will be thrown         |
+| `AssetStore::CONFLICT_OVERWRITE`    | The existing file will be replaced  |
+| `AssetStore::CONFLICT_RENAME`       | The backend will choose a new name. |
+| `AssetStore::CONFLICT_USE_EXISTING` | The existing file will be used      |
+
+If no conflict resolution scheme is chosen, or an unsupported one is requested, then the backend will choose one.
+The default asset store supports each of these.
+
+The conflict resolution is passed in to the `config` argument like so:
+
+```php
+use SilverStripe\Assets\File;
+use SilverStripe\Assets\Storage\AssetStore;
+
+// Store a file named "example-file.txt".
+$fileRecord = File::create();
+$fileRecord->setFromString(
+    'This is some file content',
+    'example-file.txt',
+    // If a file with that name already exists, let the file store rename this one.
+    config: ['conflict' => AssetStore::CONFLICT_RENAME]
+);
+$fileRecord->write();
+```
+
+See [file storage](./file_storage/) for more details about the way files are stored.
+
+## Accessing files via PHP
+
+As with storage, there are also different ways of loading the content (or properties) of the file:
+
+| Method                   | Description                                                |
+| ------------------------ | ---------------------------------------------------------- |
+| `File::getStream`        | Will get an output stream of the file content              |
+| `File::getString`        | Gets the binary content                                    |
+| `File::getURL`           | Gets the URL for this resource. May or may not be absolute |
+| `File::getAbsoluteURL`   | Gets the absolute URL to this resource                     |
+| `File::getMimeType`      | Get the mime type of this file                             |
+| `File::getMetaData`      | Gets other metadata from the file as an array              |
+| `File::getFileType`      | Return the type of file for the given extension            |
+
+### Additional file types
+
+Silverstripe CMS has a pre-defined list of common file types. `File::getFileType` will return "unknown" for files outside that list.
+
+You can add your own file extensions and its description with the following configuration.
+
+```yml
+SilverStripe\Assets\File:
+  file_types:
+    ai: 'Adobe Illustrator'
+    psd: 'Adobe Photoshop File'
+```
+
+## Renaming and moving files
+
+In order to move or rename a file you can simply update the `Name` property, or assign the `ParentID` to a new
+folder. Please note that these modifications are made simply on the draft stage, and will not be copied
+to live until a publish is made via the CMS (either on this object, or cascading from a parent).
```

Comment:
```suggestion
to live stage until a publish is made via the CMS (either on this object, or cascading from a parent).
```

Reply:
That was already the wording being used - I'm not going to change it in this PR. The refactoring was just done so that there's a sensible place for the new documentation to live.